### PR TITLE
Imporve ability to view usage data during training

### PR DESCRIPTION
**Pull request by wlf-darkmatter**
## Motivation

The wrong labeling of the dataset may cause problems such as gradient explosion during the training process. The wrong labeling can be found by setting up a hook function.

## Problem

However, when a for loop is used to traverse the iterator of the pytorch `DataLoader` class, the `hook` cannot obtain the information of the currently read image, and thus cannot determine the source of the error.

## Modification

Load the data_batch information into the `runner` during the train process, and then pass it to the hook function to solve the problem.

It's just a very small imporvement.

### CODE REVIEW

**Comment by zhouzaida** [comment]
Hi, @wlf-darkmatter , thanks for your contribution. Please also update https://github.com/open-mmlab/mmcv/blob/a86e807105d28a58565fa3a199a1f8eb47e1c548/mmcv/runner/epoch_based_runner.py#L48 and https://github.com/open-mmlab/mmcv/blob/a86e807105d28a58565fa3a199a1f8eb47e1c548/mmcv/runner/iter_based_runner.py#L76

**Comment by wlf-darkmatter** [comment]
Thanks, I have updated the files you suggested, please check.

File: mmcv/runner/epoch_based_runner.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def train(self, data_loader, **kwargs):
         self.call_hook('before_train_epoch')
         time.sleep(2)  # Prevent possible deadlock during epoch transition
         for i, data_batch in enumerate(self.data_loader):
+            self.data_batch = data_batch
```
Comment (zhouzaida) [inline mmcv/runner/epoch_based_runner.py:48]: How about deleting `self.data_batch` in line 53 to strict the scope of `runner.data_batch`

```
del self.data_batch
```

Comment (zhouzaida) [inline mmcv/runner/epoch_based_runner.py:48]: How about deleting `self.data_batch` in line 53 to strict the scope of `runner.data_batch`

```
del self.data_batch
```

Comment (wlf-darkmatter) [reply mmcv/runner/epoch_based_runner.py:48]: Ok, just saw the message, I've made changes

**Review by zhouzaida — APPROVED** [review]

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @wlf-darkmatter ！First of all, we want to express our gratitude for your significant PR in the mmcv project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤
